modules/auxiliary/gather/ldap_query.rb

```diff
@@ -184,6 +184,103 @@ def generate_rex_tables(entries, format)
     end
   end
 
+  # Decode a binary string containing a GUID into a GUID string
+  # of the format XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+  def decode_guid_binstring(guid)
+    # See https://docs.microsoft.com/en-us/dotnet/api/system.guid.tobytearray?view=netcore-3.1#remarks
+    # for more information. Specifically note that they state that the first four byte group and the
+    # following 2 groups of two bytes each are reversed. Following this, the next 2 byte group and
+    # the terminating 4 byte group are kept in the same order as in the binary data.
+    #
+    # Thanks to Spencer McIntyre for providing assistance on this code.
+    decoded_guid = [guid[0...4].reverse, guid[4...6].reverse, guid[6...8].reverse, guid[8...10], guid[10..]].map { |p| p.unpack1('H*') }.join('-')
+    decoded_guid = decoded_guid.upcase # All GUID strings must be uppercase
+
+    decoded_guid
+  end
+
+  def convert_nt_timestamp_to_time_string(nt_timestamp)
+    Time.at((nt_timestamp.to_i - 116444736000000000) / 10000000).utc.to_s
+  end
+
+  def convert_pwd_age_to_time_string(timestamp)
+    seconds = (timestamp.to_i / -1) / 10000000 # Convert always negative number to positive then convert to seconds from tick count.
+    days = seconds / 86400
+    hours = (seconds % 86400) / 3600
+    minutes = ((seconds % 86400) % 3600) / 60
+    real_seconds = (((seconds % 86400) % 3600) % 60)
+    return "#{days}:#{hours.to_s.rjust(2, '0')}:#{minutes.to_s.rjust(2, '0')}:#{real_seconds.to_s.rjust(2, '0')}"
```

Comment:
Unfortunately looks like this would also require us potentially creating a DateTime object which would require us specifying the year since a DateTime object is a set point in time, not a relative time construct like this object is.